listener/Dockerfile

```diff
@@ -0,0 +1,199 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+# Description:
+#   Builds the environment needed to build Avalon.
+#
+#  Configuration (build) parameters
+#  - proxy configuration: https_proxy http_proxy ftp_proxy
+#  - sgx mode:
+#
+# Build:
+#   $ docker build docker -f docker/Dockerfile.tcf-dev -t tcf-dev
+#   if behind a proxy, you might want to add also below options
+#   --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy --build-arg ftp_proxy=$ftp_proxy
+#
+# Run:
+#   $ cd avalon
+#   $ docker run -v $(pwd):/project/avalon -it avalon-listener /bin/bash
+#   (to run with SGX HW, add options '--device=/dev/isgx -v /var/run/aesmd:/var/run/aesmd ')
+#   make clean && make
+
+# -------------=== listener build ===-------------
+
+FROM ubuntu:bionic
+
+# Ignore timezone prompt in apt
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Add necessary packages
+RUN apt-get update \
+ && apt-get install -y -q \
+    build-essential \
+    ca-certificates \
+    cmake \
+    git \
+    make \
+    pkg-config \
+    cmake \
+    libprotobuf-dev \
+    python3-dev \
+    python3-pip \
+    python3-toml \
+    python3-requests \
+    python3-colorlog \
+    python3-twisted \
+    swig \
+    software-properties-common \
+    curl \
+    dh-autoreconf \
+    ocaml \
+    wget \
+    xxd \
+    unzip \
+    tar \
+    ocamlbuild \
+    python3-tk \
+    python3-pycodestyle \
+    pycodestyle \
+    pylint3 \
+    cppcheck \
+ && apt-get -y -q upgrade \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Install setuptools packages using pip because
+# these are not available in apt repository.
+RUN pip3 install --upgrade setuptools json-rpc nose2
+
+# Make Python3 default
+RUN ln -sf /usr/bin/python3 /usr/bin/python
+
+# SGX common library and SDK are installed in /opt/intel directory.
+# Installation of SGX libsgx-common packages requires /etc/init directory. In docker image this directory doesn't exist.
+# Hence creating /etc/init directory.
+RUN mkdir -p /opt/intel \
+ && mkdir -p /etc/init
+WORKDIR /opt/intel
+
+# Install SGX common library
+RUN wget https://download.01.org/intel-sgx/linux-2.3.1/ubuntu18.04/libsgx-enclave-common_2.3.101.46683-1_amd64.deb \
+ && dpkg -i libsgx-enclave-common_2.3.101.46683-1_amd64.deb
+
+# Install SGX SDK
+RUN wget https://download.01.org/intel-sgx/linux-2.3.1/ubuntu18.04/sgx_linux_x64_sdk_2.3.101.46683.bin \

```

Comment:
Do we need to install SGX SDK in listener?

Reply:
Yes as of now. There is another PR (in progress) to remove this dependency.

Reply:
SGX dependency can be removed now